### PR TITLE
RestoreDashboards: Create FE feature toggle

### LINKED ISSUE #89527: RestoreDashboards: Create FE feat toggle and divide functionalities

### PR DESCRIPTION
**Pull request by eledobleefe**
**What is this feature?**
This PR creates the `restoreDashboards` feature toggle.

**Why do we need this feature?**
To place the frontend work behind this feature toggle.

**Who is this feature for?**
Everyone

**Which issue(s) does this PR fix?**:

Fixes #89527 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/featuremgmt/registry.go [hunk 1]
```diff
@@ -1353,6 +1353,13 @@ var (
 			Owner:           grafanaAlertingSquad,
 			RequiresRestart: true,
 		},
+		{
+			Name:        "restoreDashboards",
```
Comment (joshhunt) [inline pkg/services/featuremgmt/registry.go:1357]: we already have `dashboardRestore` which we're using for the backend. We should be more explicit about this new feature flag that's just for the frontend.

Comment (eledobleefe) [reply pkg/services/featuremgmt/registry.go:1357]: yeah, I've just found that. On my way to change it. I will also add that it is frontend only to the description 👌

Comment (joshhunt) [reply pkg/services/featuremgmt/registry.go:1357]: Make sure that the old name doesn't persist in the `pkg/services/featuremgmt/toggles_gen.json` file

Comment (eledobleefe) [reply pkg/services/featuremgmt/registry.go]: I thought this was automatic after running the tests again. Should I just remove it?

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 2]
```diff
@@ -191,6 +191,7 @@ Experimental features might be changed or removed without prior notice.
...
 | `databaseReadReplica`                       | Use a read replica for some database queries.                                                                                                                                                                                                                                     |
 | `alertingApiServer`                         | Register Alerting APIs with the K8s API server                                                                                                                                                                                                                                    |
+| `dashboardRestoreUI`                        | Enables the frontend to be able to restore a recently deleted dashboard                                                                                                                                                                                                           |
```
Comment (L-M-K-B) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:194]: in line 187: Can you please add (backend only) in the description?

File: pkg/services/featuremgmt/registry.go [hunk 3]
```diff
@@ -1353,6 +1353,13 @@ var (
 			Owner:           grafanaAlertingSquad,
 			RequiresRestart: true,
 		},
+		{
```
Comment (L-M-K-B) [inline pkg/services/featuremgmt/registry.go:1356]: from line 1238: 
- please add (backend only) in the description
- please replace the owner by `grafanaBackendGroup`

**Review by L-M-K-B — COMMENTED** [review]
As we only want to show the new page if the feat toggle is enabled can you please change the condition in the following files:
- https://github.com/grafana/grafana/blob/6773f7f4ce0c72fa75926d6c6b248d8f84bb73a4/public/app/routes/routes.tsx#L437
- https://github.com/grafana/grafana/blob/6773f7f4ce0c72fa75926d6c6b248d8f84bb73a4/pkg/services/navtree/navtreeimpl/navtree.go#L357

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md (hunk 2 above)
Comment (eledobleefe) [reply docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:194]: I'll add it in `registry.go` as the rest of the files are auto generated 👌

File: pkg/services/featuremgmt/registry.go (hunk 3 above)
Comment (joshhunt) [reply pkg/services/featuremgmt/registry.go:1356]: Search and Storage

**Review by L-M-K-B — APPROVED** [review]
Works great!

**Pull request merged**